applications/crossbar/src/api_util.erl

```diff
@@ -715,24 +715,60 @@ prefer_new_context([{'halt', Context1}|_], Req, _) ->
     lager:debug("authn halted"),
     ?MODULE:halt(Req, Context1).
 
+
 -spec get_auth_token(cowboy_req:req(), cb_context:context()) ->
                             {cowboy_req:req(), cb_context:context()}.
 get_auth_token(Req0, Context) ->
     case cowboy_req:header(<<"x-auth-token">>, Req0) of
         {'undefined', Req1} ->
             case cb_context:req_value(Context, <<"auth_token">>) of
-                'undefined' ->
-                    lager:debug("no auth token found"),
-                    {Req1, Context};
+                'undefined' -> get_authorization_token(Req1, Context);
                 Token ->
                     lager:debug("using auth token found"),
-                    {Req1, cb_context:set_auth_token(Context, Token)}
+                    {Req1, cb_context:setters(Context, [{fun cb_context:set_auth_token/2, Token}
+                                                        ,{fun cb_context:set_auth_token_type/2, 'x-auth-token'}
+                                                       ]
+                                             )}
             end;
         {Token, Req1} ->
             lager:debug("using auth token from header"),
-            {Req1, cb_context:set_auth_token(Context, Token)}
+            {Req1, cb_context:setters(Context, [{fun cb_context:set_auth_token/2, Token}
+                                                ,{fun cb_context:set_auth_token_type/2, 'x-auth-token'}
+                                               ]
+                                     )}
     end.
 
+-spec get_authorization_token(cowboy_req:req(), cb_context:context()) ->
+                            {cowboy_req:req(), cb_context:context()}.
+get_authorization_token(Req0, Context) ->
+    case cowboy_req:header(<<"authorization">>, Req0) of
+        {'undefined', Req1} ->
+            case cb_context:req_value(Context, <<"authorization">>) of
+                'undefined' ->
+                    lager:debug("no auth token found"),
+                    {Req1, Context};
+                Authorization ->
+                    {Token, TokenType} = get_authorization_token_type(Authorization),
```

Comment:
This clause should be its own function since its duplicated in the parent case block.